test/test.js

```diff
@@ -1,35 +1,461 @@
+                                    href:
+                                        "coap://localhost:8082/TestServient/ac/setTestObject",
+                                    contentType: "application/json",
+                                    op: "invokeaction",
+                                },
+                            ],
+                            idempotent: false,
+                            safe: false,
+                        },
+                        setTestArray: {
+                            input: {
+                                type: "array",
+                                items: {
+                                    type: "number",
+                                },
+                            },
+                            output: {
+                                type: "array",
+                                items: {
+                                    type: "number",
+                                },
+                            },
+                            forms: [
+                                {
+                                    href:
+                                        "http://localhost:8081/TestServient/actions/setTestArray",
+                                    contentType: "application/json",
+                                    op: ["invokeaction"],
+                                    "htv:methodName": "POST",
+                                },
+                                {
+                                    href:
+                                        "coap://localhost:8082/TestServient/ac/setTestArray",
+                                    contentType: "application/json",
+                                    op: "invokeaction",
+                                },
+                            ],
+                            idempotent: false,
+                            safe: false,
+                        },
+                    },
+                    events: {
+                        onChange: {
+                            type: "number",
+                            forms: [
+                                {
+                                    href:
+                                        "http://localhost:8081/TestServient/events/onChange",
+                                    contentType: "application/json",
+                                    subprotocol: "longpoll",
+                                    op: ["subscribeevent"],
+                                },
+                                {
+                                    href:
+                                        "coap://localhost:8082/TestServient/ev/onChange",
+                                    contentType: "application/json",
+                                    op: "subscribeevent",
+                                },
+                            ],
+                        },
+                    },
+                    id: "urn:uuid:4d856a78-a2ff-4dff-90c2-af42afce13e0",
```

Comment:
must make sure that this matches the produced TD. If id is not given explicitly, node-wot creates one

Reply:
Then id: `"urn:uuid:4d856a78-a2ff-4dff-90c2-af42afce13e0",` should probably be removed? PerfectThing has no id statement. Other possible solutions would be to ad an id statement to perfectThing. Which solution should i choose?

Reply:
Added id statement to perfectThing.js TD.